crates/solver/src/solver/score_computation.rs

```diff
@@ -0,0 +1,169 @@
+            - self.nmb_orders_factor * nmb_orders as f64;
+        let success_probability = 1. / (1. + exponent.exp());
+        let score = success_probability * (surplus + fees) - gas_amount * gas_price;
+        Some(U256::from_f64_lossy(score))
+    }
+}
+
+// The code for collecting the data and training the model can be found here:
+// https://github.com/cowprotocol/risk_adjusted_rewards
+// The data for each solver can be found here.
+// https://drive.google.com/drive/u/1/folders/19yoL808qkp_os3BpLIYQahI3mQrNyx5T
+#[rustfmt::skip]
+const DEFAULT_SCORE_PARAMETERS: &str = "\
+    Naive,0.5604082285267333,0.00285114179288399,0.06499875450001853,3.3987949311136787;\
+    Baseline,-0.24391894879979226,-0.05809501139187965,-0.000013222507455295696,4.27946195371547;\
+    CowDexAg,-0.9613998308805674,-0.14435150204689684,0.13923418474574772,2.7565258390467178;\
+    OneInch,-0.32674185936325467,-0.05930446215554123,-0.33031769043234466,3.144609301500272;\
+    Paraswap,-0.7815504846264341,-0.06965336115721313,0.0701725936991023,3.2617622830143453;\
+    ZeroEx,-1.399997494341399,-0.04522233479453635,0.11066085229796373,2.7150950015915676;\
+    BalancerSor,-0.7070951919365344,-0.1841886790519467,0.34189609422313544,3.6849833670945027";
+
+#[derive(Debug, Parser)]
+#[group(skip)]
+pub struct Arguments {
+    /// Parameters for the score computation for each solver.
+    /// The format is a list of semicolon separated solver parameters.
+    /// Each solver parameter is a comma separated list of parameters:
+    /// [solver name],[gas amount factor],[gas price factor],[number of orders
+    /// factor],[intercept parameter]
+    #[clap(long, env, default_value = DEFAULT_SCORE_PARAMETERS)]
+    score_parameters: ScoreParameters,
+}
+
+impl Arguments {
+    pub fn get_calculator(&self, solver: SolverType) -> Option<ScoreCalculator> {
+        self.score_parameters.0.get(&solver).cloned()
+    }
+}
+
+impl Display for Arguments {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(f, "score_parameters: {:?}", self.score_parameters)
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct ScoreParameters(HashMap<SolverType, ScoreCalculator>);
+
+impl FromStr for ScoreParameters {
+    type Err = anyhow::Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        // parse user provided parameters (or default if none are provided)
+        let user_score_parameters = parse_calculators(s)?;
+        // parse default parameters and override them with the ones provided by the user
+        let score_parameters = parse_calculators(DEFAULT_SCORE_PARAMETERS)?
+            .into_iter()
+            .map(|(solver, default_score_parameter)| {
+                (
+                    solver,
+                    user_score_parameters
+                        .get(&solver)
+                        .cloned()
+                        .unwrap_or(default_score_parameter),
+                )
+            })
```

Comment:
My suggestion would be to, instead of parsing both, you would have a default:
```
fn default_score_parameters() -> HashMap<SolverType, ScoreCalculator> {
    // ...
}
```

And then have the parse function just override the default ones. It feals weird to encode default values as strings instead of actual Rust data structures.

Reply:
Just a nit.